### DISCUSSION TITLE
Google sign in - Unable to choose a different account whenever user signs in

### DISCUSSION THREAD

**Discussion by mattfranciswork0** [nextauthjs/next-auth#2891] [Help] [answered]
**Problem**
When the user signs in, they are able to choose what gmail account they want to use. Then when they  sign out and sign in **again** - they are automatically forced to use the account that they picked **previously**. They no longer can choose what account they want to sign in with.

This issue does not appear in the nextauth demo page in:https://next-auth-example.vercel.app/

Which makes me believe that it's my code, or it might be the version of my next-auth?

**Here's the image of what the 'account picker 'looks like:**
![image](https://user-images.githubusercontent.com/22691408/135927557-3b2f6115-3452-43a1-8511-2f13d549d76a.png)

**Version**
```
        "next": "^11.0.0",
        "next-auth": "^3.27.0",
```

**Sign out functionality**
```
import { signOut } from "next-auth/client";
   <li
                    onClick={() => {
                        signOut();
                    }}
                >
                    Sign Out
                </li>`
```

**Sign In Functionality (in /signin.tsx)**

```
import { getProviders, signIn } from "next-auth/client";
import { useRouter } from "next/router";
import { getSession, useSession } from "next-auth/client";
import { useEffect } from "react";
export default function SignIn() {
    const [session, loading] = useSession();
    const router = useRouter();


    useEffect(() => {
        if (!loading && !session && router.query.provider)
            //if user is not signed in
            void signIn(
                //@ts-ignore
                router.query.provider
            );
        if (!loading && session && router.query.provider) window.close(); //if user signs in
        if (!loading && !router.query.provider) router.push("/"); //if user types /signin in URL instead of clicking the sign in buttons on the header
    }, [session, loading, router]);

    return null;
```

**[...nextauth].tsx**
```
import NextAuth from "next-auth";
import Providers from "next-auth/providers";

export default NextAuth({
    // Configure one or more authentication providers

    providers: [
        Providers.Google({
            clientId: process.env.GOOGLE_ID,
            clientSecret: process.env.GOOGLE_SECRET,
        })
   ],
    pages: {
        signIn: "/signin",
    },

});
```

**Comment by balazsorban44** [comment, marked as answer]
Federated logout is not currently supported, meaning that logging out of the app only logs the app out, not your provider. Not all providers can support that either, as far as I am aware. See #836. 

You can force your provider to always show the account chooser though. Check out the `prompt=login` authorization param:
https://www.oauth.com/oauth2-servers/authorization/requiring-user-login

**Reply by mattfranciswork0** [reply]
Thank you so much!

In regards to **prompt-login**, how can we enable it in Google API? I have tried incorporating it in the callback
![image](https://user-images.githubusercontent.com/22691408/135968637-6fa3f01c-1c3f-4b1e-91ff-ea25b230a825.png)

However this breaks the sign in flow and returns me with
![image](https://user-images.githubusercontent.com/22691408/135968803-279e411a-932e-4b78-bc70-4396d8fe422d.png)

**Reply by balazsorban44** [reply]
you don't set it in the provider settings, but on your site. You have different options.

Check out the third param of the `signIn` method. This way you have per signin button control, if you need that.

If you want to enforce that behavior everywhere, you can set this:
In v4, check the `authorization.params` provider option. https://next-auth.js.org/configuration/providers/oauth-provider#options

In v3, the `authorizationParams` option: https://next-auth.js.org/v3/configuration/providers#options

**Reply by mattfranciswork0** [reply]
Thank you for the correction,
If anyone comes across this:  I managed to do it like so! This will essentially act as the federated logout as it enforces the user to choose an account.
```
    Providers.Google({
            clientId: process.env.GOOGLE_ID,
            clientSecret: process.env.GOOGLE_SECRET,
            authorizationUrl: `https://accounts.google.com/o/oauth2/auth/authorize?response_type=code&prompt=login`,
        }),
```

That being said, I dug through the links you shared. Would a **Federated logout** be available for the github and LinkedIn provider?

**Reply by balazsorban44** [reply]
It's not exactly the same from a security standpoint, and it's good to understand that if someone visits google.com for example, they will still be logged in without proper federated logout. If it's a public computer, the above solution isn't adequate.

I am not sure if GitHub or LinkedIn supports that, you would have to dig into their docs to find out.

**Reply by mattfranciswork0** [reply]
Got it, Thank you so much for all your help @balazsorban44 !

**Reply by IshfaqAhmedProg** [reply]
Any update on this for v5? with this google keeps showing the consent form everytime any user tries to login! I am running it in dev server with this as my redirect URI http://localhost:3000/api/auth/callback/google

> ```
>     Providers.Google({
>             clientId: process.env.GOOGLE_ID,
>             clientSecret: process.env.GOOGLE_SECRET,
>             authorizationUrl: `https://accounts.google.com/o/oauth2/auth/authorize?response_type=code&prompt=login`,
>         }),
> ```

**Reply by marianopaulin02** [reply]
For authjs v5 you can use the following structure
```
Providers.Google({
            clientId: process.env.GOOGLE_ID,
            clientSecret: process.env.GOOGLE_SECRET,
            authorization:{
                      url: `https://accounts.google.com/o/oauth2/auth/authorize?response_type=code&prompt=login`
            }             
}),
```
